app/workers/migrations.php

```diff
@@ -0,0 +1,314 @@
+use Utopia\CLI\Console;
+use Utopia\Database\Database;
+use Utopia\Database\Document;
+use Utopia\Transfer\Destination;
+use Utopia\Transfer\Destinations\Appwrite as DestinationsAppwrite;
+use Utopia\Transfer\Resource;
+use Utopia\Transfer\Source;
+use Utopia\Transfer\Sources\Appwrite;
+use Utopia\Transfer\Sources\Firebase;
+use Utopia\Transfer\Sources\NHost;
+use Utopia\Transfer\Sources\Supabase;
+use Utopia\Transfer\Transfer;
+
+require_once __DIR__ . '/../init.php';
+
+Console::title('Migrations V1 Worker');
+Console::success(APP_NAME . ' Migrations worker v1 has started');
+
+class MigrationsV1 extends Worker
+{
+    /**
+     * Database connection shared across all methods of this file
+     *
+     * @var Database
+     */
+    private Database $dbForProject;
+
+    public function getName(): string
+    {
+        return "migrations";
+    }
+
+    public function init(): void
+    {
+    }
+
+    public function run(): void
+    {
+        $type = $this->args['type'] ?? '';
+        $events = $this->args['events'] ?? [];
+        $project = new Document($this->args['project'] ?? []);
+        $user = new Document($this->args['user'] ?? []);
+        $payload = json_encode($this->args['payload'] ?? []);
+
+        if ($project->getId() === 'console') {
+            return;
+        }
+
+
+        /**
+         * Handle Event execution.
+         */
+        if (!empty($events)) {
+            return;
+        }
+
+        $this->dbForProject = $this->getProjectDB($this->args['project']['$id']);
+
+        $this->processMigration();
+    }
+
+    /**
+     * Process Source
+     *
+     * @return Source
+     * @throws \Exception
+     */
+    protected function processSource(array $source): Source
+    {
+        switch ($source['type']) {
+            case 'firebase':
+                return new Firebase(
+                    json_decode($source['serviceAccount'], true),
+                );
+                break;
+            case 'supabase':
+                return new Supabase(
+                    $source['endpoint'],
+                    $source['apiKey'],
+                    $source['databaseHost'],
+                    "postgres",
+                    $source['username'],
+                    $source['password'],
+                    $source['port'],
+                );
+                break;
+            case 'nhost':
+                return new NHost(
+                    $source['subdomain'],
+                    $source['region'],
+                    $source['adminSecret'],
+                    $source['database'],
+                    $source['username'],
+                    $source['password'],
+                    $source['port'],
+                );
+                break;
+            case 'appwrite':
+                return new Appwrite($source['projectId'], str_starts_with($source['endpoint'], 'http://localhost/v1') ? 'http://appwrite/v1' : $source['endpoint'], $source['apiKey']);
```

Comment:
how does the http://appwrite resolve? Don't we have to register this with the system ?

Reply:
It resolves internally within the Docker, we have to rewrite it since localhost doesn't work